src/map/zone_entities.cpp

```diff
@@ -488,23 +489,39 @@ uint16 CZoneEntities::GetNewCharTargID()
 void CZoneEntities::AssignDynamicTargIDandLongID(CBaseEntity* PEntity)
 {
     // NOTE: 0x0E (entity_update) entity updates are valid for 0 to 1023 and 1792 to 2303
-    uint16 targid = 0x700;
-    for (auto it : dynamicTargIds)
+    // Step targid up linearly from 0x700 one by one to 0x8FF unless that ID is already occupied.
+    uint16 targid = lastDynamicTargID + 1;
```

Comment:
You can use postfix ++ here, or two lines to assign and then increment

Reply:
Updated the algorithm to be "next" dynamic targID so there's no jank like this, the search will prove what the end result targid is anyway